### PR TITLE
chore: drop node version 12

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**
chore.
Drop node v12 as suggested by @alexander-akait 

**Did you add tests for your changes?**
Nope

**If relevant, did you update the documentation?**
Nope

**Summary**

**Does this PR introduce a breaking change?**

**Other information**
## https://github.com/webpack/webpack-cli/pull/3243#issuecomment-1127688694

### CODE REVIEW

**Comment by snitin315** [comment]
I suppose we should point it to the next branch.

**Comment by rishabh3112** [comment]
> I suppose we should point it to the `next` branch.

I don't think so as `next` branch is pretty outdated to qualify for a major release (290+ commits behind).
Also, as per @alexander-akait 's comment above, we need a major release now asap. So, `master` should be the location.

**Comment by snitin315** [comment]
That branch is indeed far behind. Then, should we do a release before merging this PR to master? WDYT @alexander-akait

File: package.json
Comment (snitin315) [inline package.json:11]: Let's make it 14.15.0 first LTS of node v14. This is what we are doing in all loaders and plugins.

**Review by evenstensberg — DISMISSED** [review]

File: package.json
Comment (rishabh3112) [reply package.json:11]: Yeah, sounds good.

Comment (rishabh3112) [inline package.json:11]:
```suggestion
    "node": ">=14.15.0"
```

Comment (rishabh3112) [reply package.json]: Done

**Comment by alexander-akait** [comment]
I think `next` is outdated, but there is simple list of improvements https://github.com/webpack/webpack-cli/pull/2879, so I think we can start new branch `next` (rename old `next` to `next-old`), move already done work to new `next` + solve TODOs from list, here little bit more work for us

**Comment by alexander-akait** [comment]
PR was closed because I rename `next` to `next-old`, I will not remove branch because we need to move code here.

So let's start the new `next` and merge it in `next`

**Comment by rishabh3112** [comment]
> I think next is outdated, but there is simple list of improvements https://github.com/webpack/webpack-cli/pull/2879, so I think we can start new branch next (rename old next to next-old), move already done work to new next + solve TODOs from list, here little bit more work for us

On it 👍🏽

**Comment by snitin315** [comment]
Let's also not merge any more PRs to master, to avoid rebasing.

**Comment by alexander-akait** [comment]
@snitin315 I think merging deps will not be a problem, rebasing it not hard here, but other PRs - yes

**Review by alexander-akait — APPROVED** [review]

**Pull request merged**
